exercises/primzahlen.py

```diff
@@ -0,0 +1,26 @@
+from time import *
+prim = [1, ]
+
+
+def main():
+    print("HINWEIS: Die Primzahlen von 1 bis 100'000, mit diesem Code zu berechnen dauert ca. 1079 Sekunden (17:59.147),")
+    print("die Primzahlen von 1 bis 10'000 jedoch brauchen lediglich ca. 11 Sekunden (00:11.274).")
+    N = int(input('Die Primzahlen von 1 bis wieviel willst du berechnet haben? '))
+    start_time = time()
+    for i in range(N):
+        zähler = 0
+        for w in range(N):
+            x = ((i + 1) % (w + 1))
+            if x != 0:
+                zähler = zähler + 1
+        if zähler == (N - 2):
+            print(i + 1, " - ", round((i / N) * 100, 2), "%")
+            prim.append(i + 1)
```

Comment:
Schöner Algorithmus! Allenfalls könnten Sie sich das "+1" rechnen wie folgt auch ersparen:

```suggestion
    for i in range(1, N + 1): # erzeugt eine iterierbare Liste von 1 bis und mit N
        zähler = 0
        for w in range(1, N + 1): # erzeugt eine iterierbare Liste von 1 bis und mit N
            x = i % w
            if x != 0:
                zähler = zähler + 1
        if zähler == (N - 2):
            print(i + 1, " - ", round((i / N) * 100, 2), "%")
            prim.append(i + 1)
```